### DISCUSSION TITLE
Is it possible to set the value of a variable as a span tag via `//dd:span`?

### DISCUSSION THREAD

**Discussion by your-diary** [DataDog/orchestrion#454] [Q&A] [answered]
According to the [official documentation](https://docs.datadoghq.com/tracing/trace_collection/automatic_instrumentation/dd_libraries/go/?tab=compiletimeinstrumentation#create-custom-trace-spans), one can manually create a span via a `//dd:span` comment:

```go
//dd:span custom_tag:tag_value
func CriticalPathFunction() {
  // ... implementation details ...
}
```

I expected `tag_value` in `custom_tag:tag_value` would be replaced by the value of `tag_value` variable defined in the function body, but actually it turned out it was literally interpreted.

So, in my case,

```go
//dd:span user_id:userID ret:ret
func repository(userID string) (ret string) {
	ret = "Mike"
	return
}
```

gives this result:

![2024-12-06_11_58_05](https://github.com/user-attachments/assets/c2ebb406-872f-4992-9205-24ff4485c779)

Is it possible to set the value of a variable (e.g. function paremeters) as a span tag?

**Comment by RomainMuller** [comment, marked as answer]
Hey! Not currently; but we plan on adding this feature in the future!

**Reply by korECM** [reply]
@RomainMuller 
Would you be so kind as to provide, even if only approximately, an estimate of when this feature might be implemented?

**Reply by darccio** [reply]
@korECM We don't have a timeline for this, but we'll push it higher in the priority queue.

**Comment by giou-k** [comment]
Hello, Is this feature working now?

**Reply by darccio** [reply]
Not yet, @giou-k. We'll try to prioritize it.
